### PR TITLE
Improve type detection

### LINKED ISSUE #20: "ValueError: Invalid shader code" due to additional whitespaces
during some testing for #19 I run into `ValueError: Invalid shader code.` Now this does also show up in #15 but I checked most of those cases and they were indeed correctly invalid. These wouldn't even work on the website. But the checks in #15 use a filtered dataset to only try shadertoys that include a 

But these cases are different, one example: https://www.shadertoy.com/view/3dGXDt
on the image tab, L223 has `void	mainImage` but the [shader_type method](https://github.com/pygfx/shadertoy/blob/ae33f484e8959735f241a1c733be920daf870d68/wgpu_shadertoy/shadertoy.py#L394C1-L394C13) is explicitly looking for a string with a single white space to make sure it's glsl.

So the lazy fix would be to avoid the whole elif block and just default to glsl if it isn't wgsl. Shaders without the proper `mainImage` function will raise an error instead about missing entry point perhaps.

Other ideas would be some regex (which can catch comments, as the current implementation might) or AST parsing for a fitting function. But I feel like the whole wgpu pipeline will do most of that for us already. And it's not really the place, since the whole task of that function is to decide between wgsl and glsl.

Will test around a bit and hopefully submit a PR later today.

### PR DESCRIPTION
**Pull request by Vipitis**
fixes #20 

Adds additional tests.
Uses regex to decide between wgsl and glsl.
Improved error message.

Since this is rather basic regex, this will also find functions within comments. So edge cases might still be left.

### CODE REVIEW

**Comment by Korijn** [comment]
You could search with both regexes and if both match, pick the one with the first index. At least that gives some control to sharer authors, instead of always letting wgsl win in this scenario

**Comment by almarklein** [comment]
I suspect just counting the presence of "f32" vs "float" would be a reliable measure that makes little assumptions about the actual shader.

**Comment by Korijn** [comment]
The reasonably predictable problem case here is when someone is porting GLSL to WGSL or the other way around and including the ported code in code comments. :)

**Comment by Vipitis** [comment]
two lazy ideas:

1. make `type="glsl"` a keyword argument with default. So wgsl can be specified. 
2. try both variants and expect and error in `_prepare_render()`

If the user submitted code doesn't include either entry function and we don't give them an error. The error from wgpu will be a Parse error saying function `shader_main` not found. Which isn't really helpful (hijacking and rephrasing error messages is a long term goal)

**Comment by Korijn** [comment]
Keyword argument sounds great.

**Comment by almarklein** [comment]
FWIW I was just thinking out loud with regard to distinguishing between wgsl and glsl in general. In wgpu-py we check for "#version#  for glsl and "@vertex/@fragment/@compute" for wgsl, which is also not ideal. In the end, all these are measures of confidence that can fail in specific situations.

Anyway ... if for this particular case (shadertoy shaders) detecting`void shader_main` etc. is effective, I'd say go for it.

Having a keyword argument would be nice regardless. The default could also be auto-detect. Then you'd have convenience, but also a way for the user to override in case the autodetection fails.

File: wgpu_shadertoy/shadertoy.py [hunk 1]
```diff
@@ -321,7 +321,8 @@ class Shadertoy:
 
     Parameters:
         shader_code (str): The shader code to use.
-        resolution (tuple): The resolution of the shadertoy.
+        resolution (tuple): The resolution of the shadertoy. Defaults to (800, 450).
+        shader_type (str): Can be "wgsl" or "glsl". If not provided, will be detected automatically detected.
```
Comment (Korijn) [inline wgpu_shadertoy/shadertoy.py]: Detected is double

Comment (Vipitis) [reply wgpu_shadertoy/shadertoy.py]: should be fixed in latest commit
